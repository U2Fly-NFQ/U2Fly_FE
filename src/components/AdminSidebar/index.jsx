import { Col, Menu, Row } from 'antd'
import React, { useState, useEffect } from 'react'
import { Link, useLocation } from 'react-router-dom'

const getItem = (label, key, icon, children) => {
  return {
    key,
    icon,
    children,
    label,
  }
}

function AdminSidebar({ collapsed, setCollapsed }) {
  const location = useLocation()
  const [activeLocation, setActiveLocation] = useState()

  useEffect(() => {
    setActiveLocation(location.pathname)
  }, [])

  const handleChangeKey = (item) => {
    setActiveLocation(item.key)
  }

  const Items = [
    getItem(
      <Link to="/admin">Dashboard</Link>,
      '/admin',
      <i className="fa-solid fa-gauge"></i>
    ),
    getItem(
      <Link to="/admin/discount">Discount</Link>,
      '/admin/discount',
      <i className="fa-solid fa-tags"></i>
    ),
  ]

  return (
    <>
      <Row className="admin-sidebar-header">
        <Col span={24} className="admin-sidebar-header-logo">
          <Link
            to="/"
            style={{
              fontSize: 22,
            }}
          >
            U2Fly
          </Link>
        </Col>
      </Row>
      <Menu
        theme="light"
        mode="inline"
        items={Items}
        selectedKeys={[activeLocation]}
        onClick={handleChangeKey}
      />
    </>
  )
}

export default AdminSidebar
