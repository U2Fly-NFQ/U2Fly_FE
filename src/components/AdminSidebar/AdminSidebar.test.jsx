import renderer from 'react-test-renderer'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import AdminSidebar from './'

describe('Test snapshot', () => {
  it('Should match snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <AdminSidebar />
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
