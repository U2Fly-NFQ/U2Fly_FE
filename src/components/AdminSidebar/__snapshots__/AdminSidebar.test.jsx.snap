// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Should match snapshot 1`] = `
<aside
  className="ant-layout-sider ant-layout-sider-dark admin-sidebar"
  style={
    Object {
      "flex": "0 0 200px",
      "maxWidth": "200px",
      "minWidth": "200px",
      "width": "200px",
    }
  }
>
  <div
    className="ant-layout-sider-children"
  >
    <div
      className="ant-row admin-sidebar-header"
      style={Object {}}
    >
      <div
        className="ant-col ant-col-24 admin-sidebar-header-logo"
        style={Object {}}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            src="logo.png"
          />
        </a>
      </div>
    </div>
    <ul
      className="ant-menu ant-menu-root ant-menu-inline ant-menu-light"
      data-menu-list={true}
      onKeyDown={[Function]}
      role="menu"
      tabIndex={0}
    >
      <li
        className="ant-menu-item"
        data-menu-id={null}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        tabIndex={-1}
      >
        <i
          className="fa-solid fa-gauge ant-menu-item-icon"
        />
        <span
          className="ant-menu-title-content"
        >
          <a
            href="/admin"
            onClick={[Function]}
          >
            Dashboard
          </a>
        </span>
      </li>
      <li
        className="ant-menu-item"
        data-menu-id={null}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        tabIndex={-1}
      >
        <i
          class="fa-duotone fa-badge-dollar"
          className="ant-menu-item-icon"
        />
        <span
          className="ant-menu-title-content"
        >
          <a
            href="/admin/discount"
            onClick={[Function]}
          >
            Discount
          </a>
        </span>
      </li>
    </ul>
    <div
      aria-hidden={true}
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</aside>
`;
